### PR TITLE
sys, phydat: omit scale to string conversion if not applicable 

### PR DESCRIPTION
**Pull request by smlng**
~~based on #7193~~

### CODE REVIEW

**Comment by kaspar030** [comment]
> based on #7193

please state:

- what does this PR do
- how does it do it

**Comment by smlng** [comment]
This PR adapts scale to string conversion to work with multi dimensional units like square (m^2) and cubic meters (m^3). For further info see discussion in #7193

**Comment by miri64** [comment]
#7193 was merged, please rebase.

**Comment by smlng** [comment]
rebased

**Comment by smlng** [comment]
another option would be to leave `scale_to_str` as is and make the check and rescaling for certain units in the calling function `phydat_dump`.

**Comment by miri64** [comment]
> another option would be to leave scale_to_str as is and make the check and rescaling for certain units in the calling function phydat_dump.

Yes, alternatively this check can be done in `phydat_dump()`. What do you think @haukepetersen?

**Comment by smlng** [comment]
would be less intrusive, too - I guess

**Comment by smlng** [comment]
I changed the handling as suggested

**Comment by miri64** [comment]
> I changed the handling as suggested 

Now the PR needs a new title ;-)

**Comment by smlng** [comment]
depends: which version do you and @haukepetersen like better?

File: sys/phydat/phydat_str.c [hunk 1]
```diff
@@ -32,8 +32,22 @@ void phydat_dump(phydat_t *data, uint8_t dim)
...
+        char scale_str;
+        switch (data->unit) {
+            case UNIT_UNDEF:
+            case UNIT_NONE:
+            case UNIT_M2:
+            case UNIT_M3:
+            case UNIT_PERCENT:
+            case UNIT_TEMP_C:
+            case UNIT_TEMP_F:
+            case UNIT_TEMP_K:
```
Comment (miri64) [inline sys/phydat/phydat_str.c]: Afaik Kelvin (a SI unit) my be used with SI prefix.

Comment (miri64) [inline sys/phydat/phydat_str.c]: (see e.g. https://en.wikipedia.org/wiki/Orders_of_magnitude_(temperature))

Comment (smlng) [inline sys/phydat/phydat_str.c]: though completely unrealistic for RIOT, but fine - I'll remove it

Comment (miri64) [inline sys/phydat/phydat_str.c]: Well it's one check less saving a few byte of ROM ;-)

**Comment by haukepetersen** [comment]
I don't like to put it into `phydat_dump`, as the `phydat_scale_to_str()` is/will be used also in other contexts. So I really prefer the 'white-listing' approach for simply returning `\0` for a number of units.

**Comment by smlng** [comment]
@haukepetersen currently `phydrat_scale_to_str()` simply does what it says, without any dependencies, which I think is good. The additional check for certain units would require to pass the unit to that function, which I think would change its purpose. 

IMHO for now its fine to have it in `phydrat_dump`, we may at an intermediate function later on - who is using scale_to_str, too?

**Comment by haukepetersen** [comment]
ok, fine with me. Then lets keep it like this.

**Comment by haukepetersen** [comment]
please squash

**Comment by smlng** [comment]
done

**Comment by haukepetersen** [comment]
Could you quickly fix the whitespace issues murdock is complaining about? Just squash the fixes right in...

**Comment by smlng** [comment]
> whitespace issues

damn, my bad - anyway, fix amended.

**Comment by haukepetersen** [comment]
Murodck is still unhappy, but the cause could be unrelated to your changes. Would you fix it anyway? Thx

**Comment by smlng** [comment]
I think I made non-commit, with actually pushing the fix - let's see what Murdock says now

**Comment by haukepetersen** [comment]
All green -> lets go

**Pull request merged**